internal/utils/make.go

```diff
@@ -292,14 +300,24 @@ func MakeDelegatedCredential(config *Config, parentSignConfig *Config, inCertPat
 		} else if curveName == "P-521" {
 			parentSigAlg = SignatureECDSAWithP521AndSHA512
 		} else {
-			return errors.New("ERROR: public key unsupported\n")
+			return 0, errors.New("public key unsupported")
 		}
 	case ed25519.PublicKey:
 		parentSigAlg = SignatureEd25519
+	case rsa.PublicKey:
```

Comment:
I'm not sure rsa is needed here. 

rsa is not supported in the draft due to:

```
When using RSA, the public key MUST NOT use the
rsaEncryption OID, as a result, the following algorithms are not
allowed for use with delegated credentials: rsa_pss_rsae_sha256,
rsa_pss_rsae_sha384, rsa_pss_rsae_sha512
```

`RSAPKCS1` is not supported in TLS 1.3 as far as I know: https://tools.ietf.org/id/draft-davidben-tls13-pkcs1-00.html

Per RFC8446:

```
RSASSA-PKCS1-v1_5 algorithms:  Indicates a signature algorithm using
      RSASSA-PKCS1-v1_5 [RFC8017] with the corresponding hash algorithm
      as defined in [SHS].  These values refer solely to signatures
      which appear in certificates (see Section 4.4.2.2) and are not
      defined for use in signed TLS handshake messages, although they
      MAY appear in "signature_algorithms" and
      "signature_algorithms_cert" for backward compatibility with
      TLS 1.2.
```

Reply:
Anyhow, what do you think @xvzcf ?

Reply:
As per discussion, I'll change the logic to keep it an option only for non-leaf certificates/DCs.